parcels/include/parcels.h

```diff
@@ -73,6 +73,65 @@ static inline ErrorCode spatial_interpolation_bilinear(double xsi, double eta, f
   return SUCCESS;
 }
 
+/* Bilinear interpolation routine for 2D grid for tracers with inverse distance weighting near land*/
+static inline ErrorCode spatial_interpolation_bilinear_invdist_land(double xsi, double eta, float data[2][2], float *value)
+{
+  int k, l, nb_land = 0, land[2][2] = {{0}};
+  float weight[2][2] = {{0.}}, w_sum = 0.;
+  // count the number of surrounding land points (assume land is where the value is close to zero)
+  for (int i = 0; i < 2; i++) {
+    for (int j = 0; j < 2; j++) {
+      if (fabs(data[i][j]) < 1e-14) {
+	    land[i][j] = 1;
+	    nb_land++;
+      }
+      else {
+	    // record the coordinates of the last non-land point
+	    // (for the case where this is the only location with valid data)
+	    k = i;
+	    l = j;
+      }
+    }
+  }
+  switch (nb_land) {
+  case 0:  // no land, use usual routine
+    return spatial_interpolation_bilinear(xsi, eta, data, value);
+  case 3:  // single non-land point
+    *value = data[k][l];
+    return SUCCESS;
+  case 4:  // only land
+    return ERROR_INTERPOLATION;
+  default:
+    break;
+  }
+  // interpolate with 1 or 2 land points
+  for (int i = 0; i < 2; i++) {
+    for (int j = 0; j < 2; j++) {
+      float distance = sqrt(pow((xsi - j), 2) + pow((eta - i), 2));
+      if (fabs(distance) < 1e-14) {
```

Comment:
Here, again, I'd prefer using `if(is_zero_flt(distance)) { ... }` to avoid extra tolerance thresholds.